adapter/internal/oasparser/model/api_yaml.go

```diff
@@ -97,17 +97,26 @@ type OperationYaml struct {
 
 // OperationPolicies holds policies of the APIM operations
 type OperationPolicies struct {
-	In    []Policy `json:"in,omitempty"`
-	Out   []Policy `json:"out,omitempty"`
-	Fault []Policy `json:"fault,omitempty"`
+	In    PolicyList `json:"in,omitempty"`
+	Out   PolicyList `json:"out,omitempty"`
+	Fault PolicyList `json:"fault,omitempty"`
 }
 
+// policyStats used to optimize and reduce loops by storing stats by calculating only once
+type policyStats struct {
+	firstIndex int
+	count      int
+}
+
+// PolicyList holds list of Polices in a flow of operation
+type PolicyList []Policy
+
 // Policy holds APIM policies
 type Policy struct {
-	PolicyName   string      `json:"policyName,omitempty"`
-	TemplateName string      `json:"templateName,omitempty"`
-	Order        int         `json:"order,omitempty"`
-	Parameters   interface{} `json:"parameters,omitempty"`
+	PolicyName string      `json:"policyName,omitempty"`
+	Action     string      `json:"-,omitempty"`
```

Comment:
```suggestion
	Action     string     
```
what if we remove this json / yaml tag

Reply:
when we use the name "-" yaml/json parser do not marshal/unmarshal it. we can remove `omitempty` here